docs/python/src/class/main.qml

```diff
@@ -0,0 +1,39 @@
+import QtQuick 2.0
+import QtQuick.Window 2.0
+import QtQuick.Controls 2.0
+
+import Generators 1.0
+
+Window 
+{
+    id: root
+    
+    width: 640
+    height: 480
+    visible: true
+    title: "Hello Python World!"
+    
+    Flow {
+        Button {
+            text: "Give me a number!"
+            onClicked: numberGenerator.giveNumber();
+        }
+        Label {
+            id: numberLabel
+            text: "no number"
+        }
+    }
+    
+    NumberGenerator {
+        id: numberGenerator
+        
+        // Signal argument names are not propagated from Python to QML, so we need to re-emit the signal
```

Comment:
Sadly, this is a known issue.
Maybe you can get some inspiration from [this example](http://code.qt.io/cgit/pyside/pyside-setup.git/tree/examples/declarative/textproperties) where we connect QML signals to a Python slot, then from Python we return a certain value that is used on the QML side.

Reply:
You might have seen how I work around it (by chaining two signals). I mention it as a workaround for a known issue atm, but I want to mention it to make it clear to the reader that this is the current state.